# Determination - Deterministic rendering environment for white-axe's music
# Copyright (C) 2024 Liu Hao <whiteaxe@tuta.io>
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.

# This patch silences the warning `Carla assertion failure: "strBuf != nullptr"`
# that occurs when initializing an internal plugin in Carla.

--- a/source/backend/plugin/CarlaPluginNative.cpp
+++ b/source/backend/plugin/CarlaPluginNative.cpp
@@ -277,7 +277,7 @@ public:
         carla_zeroStruct(fTimeInfo);
 
         fHost.handle      = this;
-        fHost.resourceDir = carla_strdup(engine->getOptions().resourceDir);
+        fHost.resourceDir = engine->getOptions().resourceDir != nullptr ? carla_strdup(engine->getOptions().resourceDir) : nullptr;
         fHost.uiName      = nullptr;
         fHost.uiParentId  = engine->getOptions().frontendWinId;
 
